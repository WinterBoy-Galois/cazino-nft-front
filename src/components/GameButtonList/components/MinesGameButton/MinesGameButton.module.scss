@import '../../../../styles/shared.scss';

.container {
  .highlight {
    position: absolute;
    left: 1rem;
    bottom: 16px;
    width: 37%;
    display: flex;
    flex-direction: column;
    align-items: center;

    &__animated {
      width: 54%;
      transform: translateY(6%) translateX(-15%);
      transition: transform 200ms ease-in-out;
    }

    &__static {
      width: 100%;
    }
  }

  &:hover {
    background-color: $footer-blue;
  }

  &:hover .highlight {
    &__animated {
      transform: translateY(-6px) translateX(-11px) scale(1.1);
    }
  }
}

.character {
  position: absolute;
  top: -5px;
  right: -5px;
  width: 54%;
  transition: width 200ms ease-in-out;

  &--with-sidebar {
    @include media-breakpoint-only(xl) {
      width: 61%;
    }
  }
}
