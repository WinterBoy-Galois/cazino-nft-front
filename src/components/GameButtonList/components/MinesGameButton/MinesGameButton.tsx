import React from 'react';
import styles from './MinesGameButton.module.scss';
import minesCharacter from '../../../../assets/images/gameButtons/mines-character-home-button.svg';
import minesShadow from '../../../../assets/images/gameButtons/mines-shadow-home-button.svg';
import minesBomb from '../../../../assets/images/gameButtons/mines-bomb.svg';

import { useTranslation } from 'react-i18next';
import GameButton from '../GameButton';
import { useStateValue } from '../../../../state';

interface IProps {
  onClick?: () => void;
}

const MinesGameButton: React.FC<IProps> = ({ onClick }) => {
  const { t } = useTranslation();
  const [
    {
      sidebar: { isOpen },
    },
  ] = useStateValue();

  return (
    <GameButton headline={t(`games.mines`)} onClick={onClick} className={styles.container}>
      <div className={styles.highlight}>
        <img src={minesBomb} alt="Bomb" className={styles.highlight__animated} />
        <img src={minesShadow} alt="Shadow" className={styles.highlight__static} />
      </div>
      <img
        src={minesCharacter}
        className={`${styles.character} ${isOpen ? styles['character--with-sidebar'] : ''}`}
        alt="Soldier"
      />
    </GameButton>
  );
};

export default MinesGameButton;
