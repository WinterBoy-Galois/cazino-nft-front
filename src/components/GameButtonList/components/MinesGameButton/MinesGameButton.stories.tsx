import React from 'react';
import { storiesOf } from '@storybook/react';
import MinesGameButton from '.';
import { action } from '@storybook/addon-actions';

storiesOf('Components/GameButtons/MinesGameButton', module).add('Mines', () => (
  <MinesGameButton onClick={action('click')} />
));
