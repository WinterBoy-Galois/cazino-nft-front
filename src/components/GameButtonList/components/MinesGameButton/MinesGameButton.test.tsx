import React from 'react';
import { render } from '@testing-library/react';
import MinesGameButton from './MinesGameButton';

describe('MinesGameButton', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<MinesGameButton />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
