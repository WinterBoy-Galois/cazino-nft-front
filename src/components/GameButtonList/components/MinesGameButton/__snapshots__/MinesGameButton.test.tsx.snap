// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinesGameButton should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container container"
      >
        <h2
          class="headline"
        >
          Mines
        </h2>
        <span
          class="subline"
        >
          Play
        </span>
        <div
          class="highlight"
        >
          <img
            alt="Bomb"
            class="highlight__animated"
            src="mines-bomb.svg"
          />
          <img
            alt="Shadow"
            class="highlight__static"
            src="mines-shadow-home-button.svg"
          />
        </div>
        <img
          alt="Soldier"
          class="character "
          src="mines-character-home-button.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container container"
    >
      <h2
        class="headline"
      >
        Mines
      </h2>
      <span
        class="subline"
      >
        Play
      </span>
      <div
        class="highlight"
      >
        <img
          alt="Bomb"
          class="highlight__animated"
          src="mines-bomb.svg"
        />
        <img
          alt="Shadow"
          class="highlight__static"
          src="mines-shadow-home-button.svg"
        />
      </div>
      <img
        alt="Soldier"
        class="character "
        src="mines-character-home-button.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
